### PR TITLE
chore: support for Postgres 11.16

### PR DESCRIPTION
**Pull request by michal-draftwise**
closes #22107

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [x] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by TheRealAmazonKendra** [comment]
@Mergifyio update

**Comment by Naumel** [comment]
Thank you for the time and submitting the change.

You are missing the snapshot updates matching the integ test change:
```
@aws-cdk/aws-rds:   CHANGED    integ.proxy 2.827s
@aws-cdk/aws-rds:       Resources
@aws-cdk/aws-rds: [~] AWS::RDS::DBInstance dbInstance4076B1EC 
@aws-cdk/aws-rds:  â””â”€ [~] EngineVersion
@aws-cdk/aws-rds:      â”œâ”€ [-] 11.15
@aws-cdk/aws-rds:      â””â”€ [+] 11.16
@aws-cdk/aws-rds:   UNCHANGED  integ.instance.lit 3.207s
@aws-cdk/app-delivery: $ cdk-build
@aws-cdk/pipelines: $ cdk-build
@aws-cdk/aws-rds:   UNCHANGED  integ.cluster-snapshot 3.171s
@aws-cdk/aws-rds: Snapshot Results: 
@aws-cdk/aws-rds: Tests:    1 failed, 14 total
@aws-cdk/aws-rds: Error: Some tests failed!
@aws-cdk/aws-rds: To re-run failed tests run: yarn integ-runner --update-on-failed
@aws-cdk/aws-rds:     at main (/codebuild/output/src827645417/src/github.com/aws/aws-cdk/packages/@aws-cdk/integ-runner/lib/cli.js:122:19)
@aws-cdk/aws-rds: Error: integ-runner exited with error code 1
@aws-cdk/aws-rds: Tests failed. Total time (20.8s) | /codebuild/output/src827645417/src/github.com/aws/aws-cdk/node_modules/jest/bin/jest.js (17.0s) | integ-runner (3.7s)
```

File: packages/@aws-cdk/aws-rds/test/proxy.integ.snapshot/aws-cdk-rds-proxy.template.json [hunk 1]
```diff
@@ -498,7 +498,7 @@
      "Ref": "dbInstanceSubnetGroupD062EC9E"
     },
     "Engine": "postgres",
-    "EngineVersion": "11.15",
+    "EngineVersion": "11.16",
```
Comment (corymhall) [inline packages/@aws-cdk/aws-rds/test/proxy.integ.snapshot/aws-cdk-rds-proxy.template.json:501]: @michal-draftwise I'm assuming that you just manually edited this file to fix the snapshot failure? I think in this case it is fine to just leave the integ test at `VER_11_15`. I don't think it is necessary to have an integration test for _every_ minor version.

**Review by corymhall — CHANGES_REQUESTED** [review]

**Review by corymhall — APPROVED** [review]

**Comment by comcalvi** [comment]
@Mergifyio backport v1-main

**Pull request merged**